File: 800/2128A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n, c;
        cin >> n >> c;

        vector<int> a(n);
        for (int i = 0; i < n; i++) 
            cin >> a[i];
        
        sort(a.begin(), a.end());

        int coins = 0;
        long long m = 1;

        for (int i = n - 1; i >= 0; i--) 
        {
            if (1LL * a[i] * m <= c) 
                m *= 2;
            else 
                coins++;
        }

        cout << coins << "\n";
    }

    return 0;
}

File: 800/2128A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int c = sc.nextInt();
            
            int a[] = new int[n];
            for(int i=0;i<n;i++)
                a[i] = sc.nextInt();
            
            Arrays.sort(a);
            int coins = 0;
            long m = 1;
            
            for(int i=n-1;i>=0;i--)
            {
                if(a[i] * m <= c)
                    m = m*2;
                else
                    coins++;
            }
            
            System.out.println(coins);
        }
    }
}